### DISCUSSION TITLE
How to disable open page in new window?

### DISCUSSION THREAD

**Discussion by OdinTemple** [cefsharp/CefSharp#3896] [General]
I search a lot, to do so is to use OnBeforePopup to handle
However, here's the problem:

At first, I tried to disable the new windows, but all I can found is this:

```
    bool ILifeSpanHandler.OnBeforePopup(*******)
    {
        newBrowser = null;
        OldBrowser.Load(targetUrl);
        return true; 
    }
```

However, it won't work well, I guess this popup includes referer and cookies, without which the redirect won't go as usual.

So I have to use this:
```

               newBrowser = new ChromiumWebBrowser(""); 
                pop = (ChromiumWebBrowser)newBrowser;
                return false; 
```

So that now I can handle the new window via the global variable pop.
However, I just can't put the pop into any panel.
When I try this:

panel1.Controls.Add(pop);

the new window just resized, but it won't go into the panel

So my question are below:
1. How to handle the redirect prefectly? The redirect is very complicated, I use fiddler, and I cought many requests that I just don't know which cause the difference between the natural popup and the manual redirect.
2. Why pop can't be add to panel, and if I insist to do so, can it be done? Is it supported in the current version?

**Comment by OdinTemple** [comment]
Even if I can simulate the redirect prefectly, I think it is really stupid to do so much works, including manually extract every step in the process, just to simulate a single click. There should be a bettter way to do such works

**Comment by amaitland** [comment]
In future please use discussions to ask questions. https://github.com/cefsharp/CefSharp/discussions

> Why pop can't be add to panel, and if I insist to do so, can it be done? Is it supported in the current version?

It is possible see https://github.com/cefsharp/CefSharp/pull/3529#issuecomment-828934239

The project source has a working example https://github.com/cefsharp/CefSharp/blob/7f870db945db29c447555724d200fed281791f6e/CefSharp.WinForms.Example/BrowserTabUserControl.cs#L55

**Comment by OdinTemple** [comment]
> In future please use discussions to ask questions. https://github.com/cefsharp/CefSharp/discussions
> 
> > Why pop can't be add to panel, and if I insist to do so, can it be done? Is it supported in the current version?
> 
> It is possible see [#3529 (comment)](https://github.com/cefsharp/CefSharp/pull/3529#issuecomment-828934239)
> 
> The project source has a working example
> 
> https://github.com/cefsharp/CefSharp/blob/7f870db945db29c447555724d200fed281791f6e/CefSharp.WinForms.Example/BrowserTabUserControl.cs#L55

thanks, but How I can use it in my project? I just couldn't use Create()...
![图片](https://user-images.githubusercontent.com/18748404/142929688-c5787bf9-00bf-452f-ae9e-411e0fe6f576.png)

**Reply by amaitland** [reply]
Please no images of code, I cannot copy and paste from an image. 

http://cefsharp.github.io/api/95.7.x/html/M_CefSharp_WinForms_Handler_LifeSpanHandler_Create.htm

**Reply by OdinTemple** [reply]
my Cefsharp version is 95.7.147, I meant, why I can't use the static method "LifeSpanHandler.Create()"? Yes new LifeSpanHandlerBuilder().Create() does works, but, where is the LifeSpanHandler.Create()? It doesn't make sense

**Reply by amaitland** [reply]
Pay attention to the namespace.

**Comment by OdinTemple** [comment]
@amaitland  I came back to say thank you that your perfectly solve my problem, I think that the `OnBeforePopup()` way is completely obsolete and should be abandoned forever...
